src/dapla_pseudo/v1/baseclasses.py

```diff
@@ -0,0 +1,303 @@
+                for request in pseudo_requests
+            ]
+
+            for future in as_completed(futures):
+                field_name, data, raw_metadata = future.result()
+                self._dataset = self._dataset.with_columns(data.alias(field_name))
+                raw_metadata_fields.append(raw_metadata)
+
+            return PseudoFieldResponse(
+                data=self._dataset, raw_metadata=raw_metadata_fields
+            )
+
+    def _pseudonymize_dataset(
+        self,
+        pseudo_request: PseudoFileRequest | DepseudoFileRequest | RepseudoFileRequest,
+        timeout: int,
+    ) -> PseudoFileResponse:
+        data_spec: FileSpecDecl
+
+        request_spec: FileSpecDecl = (
+            None,
+            pseudo_request.to_json(),
+            str(Mimetypes.JSON),
+        )
+
+        match self._dataset:
+            case File(file_handle, content_type):
+                file_name = _extract_name(
+                    file_handle=file_handle, input_content_type=content_type
+                )
+                data_spec = (
+                    file_name,
+                    file_handle,
+                    str(pseudo_request.target_content_type),
+                )
+
+                response = self._pseudo_client._post_to_file_endpoint(
+                    path=f"{self._pseudo_operation.value}/file",
+                    request_spec=request_spec,
+                    data_spec=data_spec,
+                    stream=True,
+                    timeout=timeout,
+                )
+                file_handle.close()
+            case HierarchicalDataFrame(dataset):
+                file_name = "data.json"
+                data_spec = (
+                    file_name,
+                    json.dumps(dataset.to_dicts()),
+                    str(pseudo_request.target_content_type),
+                )
+
+                response = self._pseudo_client._post_to_file_endpoint(
+                    path=f"{self._pseudo_operation.value}/file",
+                    request_spec=request_spec,
+                    data_spec=data_spec,
+                    stream=True,
+                    timeout=timeout,
+                )
+
+        payload = json.loads(response.content.decode("utf-8"))
+        pseudo_data = payload["data"]
+        metadata = RawPseudoMetadata(
+            logs=payload["logs"],
+            metrics=payload["metrics"],
+            datadoc=payload["datadoc_metadata"]["pseudo_variables"],
+        )
+
+        return PseudoFileResponse(
+            data=pseudo_data,
+            raw_metadata=metadata,
+            content_type=Mimetypes.JSON,
+            streamed=True,
+            file_name=file_name,
+        )
+
+
+class _BaseRuleConstructor:
+    """Base class for the _PseudoFuncSelector/_DepseudoFuncSelector/_RepseudoFuncSelector builders."""
+
+    def __init__(
+        self,
+        fields: list[str],
+        dataset_type: type[pl.DataFrame] | type[File] | type[HierarchicalDataFrame],
```

Comment:
HierarchicalDataFrame was removed in #383

Reply:
Aha, I didn't see this PR :) Merge went a bit quickly I see